service/telemetry/otel_trace_sampler.go

```diff
@@ -10,7 +10,7 @@ import (
 type recordSampler struct{}
 
 func (r recordSampler) ShouldSample(_ sdktrace.SamplingParameters) sdktrace.SamplingResult {
-	return sdktrace.SamplingResult{Decision: sdktrace.RecordOnly}
+	return sdktrace.SamplingResult{Decision: sdktrace.RecordAndSample}
```

Comment:
We should respect parent sampling decision and not always sample blindly.

Reply:
Let me know if I'm misunderstanding something, but I think this is what this code does. This recordSampler type is used lower down in the method called "alwaysRecord" which returns a sampler. This is used to make a tracer provider with a sampler that always decides to record.

However, if a parent span is created with this sampler, none of its children are recorded. This doesn't seem to respect the "always record" contract to me.

From a purely functionality perspective, it seems like a bug that the collector only collects telemetry spans about its receivers, and not for any other component. This means that there is little point in having traces because there's only one span in them.

Reply:
Use this https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#ParentBased

Reply:
Would it make more sense to add a line here:

https://github.com/open-telemetry/opentelemetry-collector/blob/c6b5e77f06f6d98091490927276111806a8682eb/service/telemetry/otel_trace_sampler.go#L26

like:

```
sdktrace.WithLocalParentNotSampled(sdktrace.AlwaysSample()),
```

?